client/ayon_traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -482,6 +526,22 @@ def _get_data_from_csv(
         missing_tasks: Set[str] = set()
         for product_item in product_items_by_name.values():
             folder_path = product_item.folder_path
+
+            if folder_path in missing_paths:
+                product_item.has_promised_context = True
+                product_item.task_type = None
+                try:
+                    product_item.parents = self._validate_parents(
+                        project_name,
+                        product_item
+                    )
+                except ValueError:
+                    raise CreatorError(
+                        f"Parent context must exists for new shots: {folder_path}"
```

Comment:
The code expects the parent folder to newly-created shots to exists.
I thought it was to ambigious to assume to folder type(s) for the whole hierarchy otherwise. 

Is that a good behavior ?

Reply:
I don't think it is. We don't have any logic to define folder types of parents, it is right now hardcoded (as far as I know), but they usually don't exist on first editorial publish.

Reply:
This is quite challenging. Non-existent parent types should be defined in the settings. If they match, they can be resolved. If not, let's set them as the default folder type.

  

The mapping for folder names to types could use a regex pattern input. The settings for this mapping should be added to the traypublisher CSV editorial here: `ayon+settings://traypublisher/create/IngestCSV`.

Reply:
@robin-ynput it seems you are having very old server - update it please. The list items are now looking different for some time. This is how they look now:  ![image](https://github.com/user-attachments/assets/c4226eaa-3d8c-44f2-a473-b0e6e980da11)

 Anyway, the folder types will better be defined as enumerator of available folder types from project anatomy. I will commit the recommendations here f199802fe994c90dc54f74f59cc4fc30852b2451